File: Zigzag Conversion/solve.cpp
#include <bits/stdc++.h>
using namespace std;
class Solution
{
public:
    string convert(string s, int rows)
    {

        if (rows == 1)
            return s;
        string out = "";
        int len = s.size();

        for (int i = 1; i <= rows; i++)
        {
            int rr = rows - i;
            int indx = i - 1;
            for (int j = 1;; j++)
            {
                if (indx >= len)
                    break;
                out.push_back(s[indx]);
                if (i == 1 || i == rows)
                    indx += (rows - 1) * 2;
                else if (j % 2 == 1)
                    indx += rr * 2;
                else if (j % 2 == 0)
                    indx += (i - 1) * 2;
            }
        }
        return out;
    }
};

File: ff/Zigzag Conversion/solve.js
/**
 * @param {string} s
 * @param {number} numRows
 * @return {string}
 */
var convert = function (s, rows) {
    if (rows === 1) return s;
    var out = "";
    const len = s.length;

    for (let i = 1; i <= rows; i++) {
        let rr = rows - i;
        let indx = i - 1;
        for (let j = 1; ; j++) {
            if (indx >= len) break;
            out += s[indx];
            if (i == 1 || i == rows) indx += (rows - 1) * 2;
            else if (j % 2 == 1) indx += rr * 2;
            else if (j % 2 == 0) indx += (i - 1) * 2;
        }
    }
    return out;
};
